src/transformers/models/oneformer/processing_oneformer.py

```diff
@@ -76,13 +98,41 @@ def _preprocess_text(self, text_list=None, max_length=77):
         token_inputs = torch.cat(token_inputs, dim=0)
         return token_inputs
 
-    def __call__(self, images=None, task_inputs=None, segmentation_maps=None, **kwargs):
+    @staticmethod
+    def _check_args(
+        images: Optional[ImageInput] = None,
+        task_inputs: Optional[Union[TextInput, PreTokenizedInput]] = None,
+    ):
+        if task_inputs is None:
+            raise ValueError("You have to specify the task_inputs. Found None.")
+        elif images is None:
+            raise ValueError("You have to specify the images. Found None.")
+
+        if isinstance(task_inputs, str):
+            task_inputs = [task_inputs]
+
+        if not isinstance(task_inputs, List) or not task_inputs:
+            raise TypeError("task_inputs should be a string or a list of strings.")
```

Comment:
since this does more than checking the types, I suggest moving the conversion to list out of this function, and rename it explicitly `_validate_input_types` for instance.

Reply:
Implemented.